starlette/staticfiles.py

```diff
@@ -73,19 +83,21 @@ def __init__(self, scope, path, check_directory=None):
         if not (stat.S_ISDIR(stat_result.st_mode) or stat.S_ISLNK(stat_result.st_mode)):
             raise RuntimeError("StaticFiles path '%s' is not a directory." % directory)
 
-    async def __call__(self, receive, send):
+    async def __call__(self, receive: Receive, send: Send) -> None:
         if self.check_directory is not None:
             await self.check_directory_configured_correctly()
 
         try:
             stat_result = await aio_stat(self.path)
         except FileNotFoundError:
-            response = PlainTextResponse("Not Found", status_code=404)
+            response = PlainTextResponse(
+                "Not Found", status_code=404
+            )  # type: typing.Union[PlainTextResponse, FileResponse]
```

Comment:
Can we avoid having these noisy comments? Is there an option so that mypy will allow us to not error on missing variable types?

Reply:
I get why it's there, but I don't think they add much personally. If we *do* need it then `type: Response` would be preferable.

Reply:
Yes, totally right. Should be `type: Response`. It does not add anything, but if we don't add it `Incompatible types in assignment` error is raised. I can not find any option to allow it, and I think make sense because we would be breaking the `Liskov principle`, I guess.